src/frontend/src/lib/derived/token.derived.ts

```diff
@@ -29,7 +34,20 @@ export const tokenDecimals: Readable<number | undefined> = derived(
 	([$token]) => $token?.decimals
 );
 
-export const tokenToggleable: Readable<boolean> = derived(
-	[token],
-	([$token]) => nonNullish($token) && 'enabled' in $token
-);
+export const tokenToggleable: Readable<boolean> = derived([token], ([$token]) => {
+	if (nonNullish($token)) {
+		let toggleable = false;
+
+		if (icTokenIcrcCustomToken($token)) {
+			toggleable = !isIcrcCustomTokenDisabled($token);
+		} else if (icTokenEthereumUserToken($token)) {
+			toggleable = !isEthereumUserTokenDisabled($token);
```

Comment:
just noticing, isn't it better to create the functions that check if it is enabled? since you are negating it in all usage (or not?)